python/fbprophet/diagnostics.py

```diff
@@ -105,43 +110,84 @@ def cross_validation(model, horizon, period=None, initial=None):
-                'Increase initial window.'
-            )
-        m.fit(history_c, **model.fit_kwargs)
-        # Calculate yhat
-        index_predicted = (df['ds'] > cutoff) & (df['ds'] <= cutoff + horizon)
-        # Get the columns for the future dataframe
-        columns = ['ds']
-        if m.growth == 'logistic':
-            columns.append('cap')
-            if m.logistic_floor:
-                columns.append('floor')
-        columns.extend(m.extra_regressors.keys())
-        columns.extend([
-            props['condition_name']
-            for props in m.seasonalities.values()
-            if props['condition_name'] is not None])
-        yhat = m.predict(df[index_predicted][columns])
-        # Merge yhat(predicts), y(df, original data) and cutoff
-        predicts.append(pd.concat([
-            yhat[predict_columns],
-            df[index_predicted][['y']].reset_index(drop=True),
-            pd.DataFrame({'cutoff': [cutoff] * len(yhat)})
-        ], axis=1))
+
+    if multiprocess is True:
+        with Pool() as pool:
+            logger.info('Running cross validation in multiprocessing mode')
+            input_df = [[df, model, cutoff, horizon, predict_columns] for cutoff in cutoffs]
+            predicts = pool.starmap(single_cutoff_forecast, input_df)
+    elif multiprocess is None:
+        predicts = []
+        for cutoff in cutoffs:
+            predicts.append(single_cutoff_forecast(df, model, cutoff, horizon, predict_columns))
+    else:
+        raise ValueError(f"{multiprocess} is not a valid assignment to multiprocess argument.Valid "
+                         f"options are 'None' or 'True'")
+
 
     # Combine all predicted pd.DataFrame into one pd.DataFrame
     return pd.concat(predicts, axis=0).reset_index(drop=True)
 
+
+def single_cutoff_forecast(df, m, cutoff, horizon, predict_columns):
+    """Forecast for single cutoff. Used in cross validation function
+    when evaluating for multiple cutoffs either sequentially or in parallel .
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        DataFrame with history to be used for single
+        cutoff forecast
+    m: Prophet model.
+    cutoff: string with pd.Timedelta compatible style.
+        Simulated Forecast will start from this date.
+    horizon: string with pd.Timedelta compatible style, e.g., '5 days',
+        '3 hours', '10 seconds'.
+    predict_columns: List of strings e.g. ['ds', 'yhat']
+        Columns with date and forecast to be returned in output.
+
+    Returns
+    -------
+    A pd.DataFrame with the forecast, actual value and cutoff.
+
+    """
+
+    # Generate new object with copying fitting options
+    m = prophet_copy(m, cutoff)
+    # Train model
+    history_c = df[df['ds'] <= cutoff]
+    if history_c.shape[0] < 2:
+        raise Exception(
+            'Less than two datapoints before cutoff. '
+            'Increase initial window.'
+        )
+    m.fit(history_c, **m.fit_kwargs)
```

Comment:
This actually isn't going to work - you'll notice in the original it is doing
```
m.fit(history_c, **model.fit_kwargs)
```
where `model` is the original model (`m` is the copy).
`fit_kwargs` are set during fitting, and aren't copied over from `model` to `m` by `prophet_copy`.

The purpose of this is that if a particular kwarg was used for fitting the original model (like `algorithm=Newton`) we want to use that same kwarg when fitting the CV model.

That can easily be done here by just changing the name of the arg to this function to be `model` so that we keep track of which was the original model vs. the copy, and using `model.fit_kwargs` here.

Reply:
thanks, fixed now